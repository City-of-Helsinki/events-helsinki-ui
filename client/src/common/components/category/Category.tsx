import React, { FunctionComponent } from "react";

import CloseIcon from "../../../icons/CloseIcon";
import { formatMessage } from "../../translation/TranslationUtils";
import { Category as CategoryType } from "../../types";
import styles from "./category.module.scss";

interface Props {
  category: CategoryType;
  onRemove: (category: CategoryType) => void;
}

const Category: FunctionComponent<Props> = ({ category, onRemove }) => {
  const handleRemove = () => {
    onRemove(category);
  };

  return (
    <div className={styles.category}>
      <button
        type="button"
        className={styles.closeButton}
        onClick={handleRemove}
        aria-label={formatMessage("commons.category.ariaButtonRemove", {
          category: category.text
        })}
      >
        <CloseIcon />
      </button>
      {category.text}
    </div>
  );
};

export default Category;
