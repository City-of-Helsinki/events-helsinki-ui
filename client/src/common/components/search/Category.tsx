import React, { FunctionComponent } from "react";

import { ReactComponent as CloseIcon } from "../../../assets/icons/svg/close.svg";
import { formatMessage } from "../../translation/utils";
import { Category as CategoryType } from "../../types";
import styles from "./category.module.scss";

interface Props {
  category: CategoryType;
  onRemove: (category: CategoryType) => void;
}

const Category: FunctionComponent<Props> = ({ category, onRemove }) => {
  const handleRemove = () => {
    onRemove(category);
  };

  return (
    <div className={styles.category}>
      <button
        type="button"
        className={styles.closeButton}
        onClick={handleRemove}
        aria-label={formatMessage("commons.category.ariaButtonRemove", {
          category: category.text
        })}
      >
        <CloseIcon />
      </button>
      {category.text}
    </div>
  );
};

export default Category;
