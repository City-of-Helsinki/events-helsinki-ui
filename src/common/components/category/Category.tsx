import { IconCross } from 'hds-react';
import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import { Category as CategoryType } from '../../types';
import styles from './category.module.scss';

interface Props {
  category: CategoryType;
  onRemove: (category: CategoryType) => void;
}

const Category: FunctionComponent<Props> = ({ category, onRemove }) => {
  const { t } = useTranslation();

  const handleRemove = () => {
    onRemove(category);
  };

  return (
    <div className={styles.category}>
      <button
        type="button"
        className={styles.closeButton}
        onClick={handleRemove}
        aria-label={t('commons.category.ariaButtonRemove', {
          category: category.text,
        })}
      >
        <IconCross aria-hidden />
      </button>
      {category.text}
    </div>
  );
};

export default Category;
