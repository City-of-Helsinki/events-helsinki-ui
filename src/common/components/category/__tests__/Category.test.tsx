import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import Category from '../Category';

const category = { text: 'foo', value: 'bar' };

it('matches snapshot', () => {
  const { container } = render(
    <Category category={category} onRemove={jest.fn()} />
  );

  expect(container.firstChild).toMatchSnapshot();
});

it('calls onRemove callback when remove button is clicked', () => {
  const onClickMock = jest.fn();
  render(<Category category={category} onRemove={onClickMock} />);

  expect(screen.queryByText(/foo/i)).toBeInTheDocument();

  userEvent.click(screen.getByRole('button'));

  expect(onClickMock).toHaveBeenCalled();
});
