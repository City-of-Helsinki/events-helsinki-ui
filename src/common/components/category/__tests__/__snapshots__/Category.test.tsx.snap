// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category matches snapshot 1`] = `
<div
  className="category"
>
  <button
    aria-label="Poista kategoria: foo"
    className="closeButton"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      style={Object {}}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <path
          d="M0 0h24v24H0z"
        />
        <path
          d="M17.657 7.757L13.414 12l4.243 4.243-1.414 1.414L12 13.414l-4.243 4.243-1.414-1.414L10.586 12 6.343 7.757l1.414-1.414L12 10.586l4.243-4.243z"
          fill="currentColor"
        />
      </g>
    </svg>
  </button>
  foo
</div>
`;
